src/query-builder/InsertQueryBuilder.ts

```diff
@@ -600,4 +613,19 @@ export class InsertQueryBuilder<Entity> extends QueryBuilder<Entity> {
         throw new InsertValuesMissingError();
     }
 
+    /**
+     * Checks if column is an auto-generated primary key, but the current insertion specifies a value for it.
+     * 
+     * @param column
+     */
+    protected isOverridingAutoIncrementBehavior(column: ColumnMetadata): boolean {
+        return column.isPrimary 
+                && column.isGenerated 
+                && column.generationStrategy === "increment"
+                && this.getValueSets().some((valueSet) => 
+                    column.getEntityValue(valueSet) !== undefined 
+                    && column.getEntityValue(valueSet) !== null
```

Comment:
If it is null isn't that technically a valid value that we'd want to count as a "supplied" value

Reply:
In general, `null` is not a valid value for a primary key (because that is a clustered index that determines on-disk ordering of entries). Most databases (including mssql) prevent you from doing this.

Now, one could argue that this should fail instead of falling back to auto increment. However, that would be inconsistent with common SQL behavior, as databases cannot distinguish between undefined and null, and as such would insert a auto generated primary key if you supply null in a query.

Reply:
What if it is '0'. I want to use the 'save()' method which can update and insert and when I want to insert then I set the data of identity column is 0. And it really insert 0 into the identity column. So I can not insert more data because it only update this  row.